### PR TITLE
Fix bold/italic/underline bugs. See BLUE-657.

### PR DESCRIPTION
**Pull request by internettrans**
https://canopytax.atlassian.net/browse/BLUE-657

### CODE REVIEW

File: src/rich-text-editor.component.js [hunk 1]
```diff
@@ -34,6 +34,7 @@ export const RichTextEditor = forwardRef((props, editorRef) => {
     const selection = window.getSelection()
     selection.removeAllRanges()
     selection.addRange(range)
+    document.execCommand('removeFormat')
```
Comment (internettrans) [inline src/rich-text-editor.component.js:37]: Fixes bug where font styling wasn't reset when the editor was emptied

File: src/rich-text-editor.component.js [hunk 2]
```diff
@@ -72,29 +73,31 @@ export const RichTextEditor = forwardRef((props, editorRef) => {
...
     richTextContext.selectRangeFromBeforeBlur = () => {
       if (divRef.current && document.activeElement !== divRef.current && !divRef.current.contains(document.activeElement)) {
-        if (selectionRangeBeforeBlurRef.current) {
-          const currentSelection = window.getSelection()
-          currentSelection.removeAllRanges()
-          currentSelection.addRange(selectionRangeBeforeBlurRef.current)
-        } else {
-          divRef.current.focus()
-        }
+        divRef.current.focus()
```
Comment (internettrans) [inline src/rich-text-editor.component.js:81]: calling focus just so happens to select the previously selected text from before the blur. So we actually don't need to call removeAllRanges and addRange with the last range.

Comment (dvnrsn) [reply src/rich-text-editor.component.js:81]: nice!

File: src/rich-text-editor.component.js [hunk 3]
```diff
@@ -162,10 +165,31 @@ export const RichTextEditor = forwardRef((props, editorRef) => {
   }
 })
 
+// This hook allows you to change the focused value synchronously instead of queuing it
```
Comment (dvnrsn) [inline src/rich-text-editor.component.js:168]: this is lifechanging feedback I know but `queueing`

Comment (internettrans) [reply src/rich-text-editor.component.js:168]: oh haha good catch

**Pull request merged**